src/Microsoft.ServiceFabric.Actors/Runtime/ActorService.cs

```diff
@@ -259,9 +307,17 @@ protected override IEnumerable<ServiceReplicaListener> CreateServiceReplicaListe
         /// can impact availibility of your service.
         /// </para>
         /// </remarks>
-        protected override Task RunAsync(CancellationToken cancellationToken)
+        protected override async Task RunAsync(CancellationToken cancellationToken)
         {
-            return this.ActorManager.StartLoadingRemindersAsync(cancellationToken);
+            // If Migration attibute is set to source and StateProvider is KvsActorStateProvider
+            if (this.migrationOrchestrator != null)
+            {
+                // TODO: Manual start.
+                await this.migrationOrchestrator.StartMigrationAsync(cancellationToken);
+                await this.ActorManager.StartLoadingRemindersAsync(cancellationToken);

```

Comment:
Duplicate code. we are loading reminders twice when service is migration ready.

Reply:
Added a return statement to avoid duplicate load